cron/hourly/check-pending-transferwise-transactions.js

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env node
+import '../../server/env';
+
+import models from '../../server/models';
+import statuses from '../../server/constants/expense_status';
+import * as transferwiseLib from '../../server/lib/transferwise';
+import activities from '../../server/constants/activities';
+import { PayoutMethodTypes } from '../../server/models/PayoutMethod';
+
+async function processExpense(expense) {
+  console.log(`\nProcessing expense #${expense.id}...`);
+  const host = await expense.collective.getHostCollective();
+  if (!host) {
+    throw new Error(`Could not find the host embursing the expense.`);
+  }
+  const [connectedAccount] = await host.getConnectedAccounts({
+    where: { service: 'transferwise', deletedAt: null },
+  });
+  if (!connectedAccount) {
+    throw new Error(`Host is not connected to Transferwise.`);
+  }
+  const [transaction] = expense.Transactions;
+  if (!transaction) {
+    throw new Error(`Could not find any transactions associated with expense.`);
+  }
+  const transfer = await transferwiseLib.getTransfer(connectedAccount.token, transaction.data.transfer.id);
+  if (transfer.status === 'processing') {
+    console.warn(`Transfer is still being processed, nothing to do but wait.`);
+  } else if (transfer.status === 'outgoing_payment_sent') {
+    console.log(`Transfer sent, marking expense as paid.`);
+    await expense.setPaid(expense.lastEditedById);
+    await expense.createActivity(activities.COLLECTIVE_EXPENSE_PAID);
+  } else if (transfer.status === 'funds_refunded') {
+    console.warn(`Transfer ${transfer.status}, setting status to Error and deleting existing transactions.`);
+    await models.Transaction.destroy({ where: { ExpenseId: expense.id } });
```

Comment:
Normally, the `Transaction` table is immutable.

Might be ok for a MVP but to be considered for the future.

Reply:
I agree. The problem is that we don't consider expenses when calculating the available balance of a collective, which means that while the transaction isn't created, the expense amount will be still available for the collective to use.
Since this payment process is async and can take up to a few days, that's a big issue.
Do you have any suggestions on how we could solve that?